[FIX] Fixed issues #11 and #12

gabrielo77 opened the pull request.

ssaid committed:
[FIX] point_of_sale,retentions_basic,perceptions_basic

En los xml de data, agrego el forcecreate="0" para todos los registros.
Es necesario para que en la migracion cuando tiro -u all no me tire constraint error.

Conflicts:

gabrielo77 committed:
[FIX] Fixed issues #11 and #12

sebastiken committed:
Merge pull request #10 from ssaid/8.0

[FIX] point_of_sale,retentions_basic,perceptions_basic

sebastiken commented:
@gabrielo77 primero que nada, gracias por tu aporte. Te comento un poco el tema que planteás del _domain_ que no te permite buscar _configuraciones de subdiario_. En un primer momento, se pensó este módulo para permitir hacer el reporte que uno quiera, siempre basado en **aeroo**. Entonces, un usuario podía crear un _reporte de subdiario_ con el header de tal manera y que se imprima en **pdf**. Otro quizás lo quería, pero en **xls**, etc. Por eso, desde la configuración de subdiario, te permite elegir Reporte _(de forma poco intuitiva, porque te aparecen todos los reportes del sistema, sean o no de subdiario)_ y luego elegís los _impuestos_ que querés que formen parte de ese reporte.
Hoy por hoy, la experiencia mostró que no tiene mucho sentido poder elegir entre varios reportes diferentes, y siempre terminamos utilizando el mismo reporte que es el que viene en el módulo.
Volviendo al [domain](https://github.com/OCA/l10n-argentina/blob/8.0/l10n_ar_tax_reports/wizard/account_tax_subjournal_view.xml#L17) que está en la vista, el mismo se basa en el campo **report_id**, que se llena como un _default_ en la función [_get_report_id](https://github.com/OCA/l10n-argentina/blob/8.0/l10n_ar_tax_reports/wizard/account_tax_subjournal.py#L29). Si te fijás, en esa función, busca por un nombre de **reporte** y se queda con ese. O sea, en definitiva, no solo no tiene mucho sentido permitir usar diferentes **reportes** sino que por código te obliga a usar uno solo. Es algo que nos quedó un tanto colgado y habría que subsanarlo.
Si te animás vos, te pediría que quites todo lo relacionado con esa funcionalidad de elegir el reporte. Esto tanto en el wizard, ese campo **report_id** como en el objeto _tax_subjournal_report_config_ el campo [tax_report_id](https://github.com/OCA/l10n-argentina/blob/8.0/l10n_ar_tax_reports/tax_report_config.py#L31). Eso implica dejar el _domain_ como lo pusiste en este _PR_, que no tenga en cuenta ese campo **report_id**.
Espero tus novedades, críticas, ideas y gracias.

sebastiken committed:
Modificacion ACL para wsaa_ta

[MOD] Se le permite al usuario Contable modificar el TA (Ticket de
Acceso) sino solo el manager podia hacerlo y resulta incomodo ya que
cada tanto el TA expira.

Commit:
[MIG] l10n_ar_point_of_sale->sale.py

Commit:
[RM] pbd

sebastiken committed:
Merge pull request #18 from ssaid/pos-mig

Migracion a 8.0 de l10n_ar_point_of_sale->sale.py

sebastiken committed:
Fix fe_param_get_ptos_venta, nombre key en ResultGet

sebastiken committed:
Limpieza y actualizacion de __openerp__.py en algunos modulos

sebastiken committed:
ACL en base_vat_ar

sebastiken committed:
Mocking l10n_ar_wsaa. Testing basico

sebastiken committed:
Modificaciones en testing WSAA

sebastiken committed:
Actualizacion datos de demo

sebastiken committed:
Add metodo get_server_state

sebastiken committed:
Agregados testing WSFE basico

sebastiken committed:
Fix wizard wsfe_sinchronize_voucher.

[FIX] Se mezclaba la api nueva con la vieja en el onchange del wizard y
eso hacia que no se pueda utilizar el mismo.

sebastiken committed:
Refactor wsfe_sinchronize_voucher a api nueva

[MOD] Se reescribe todo el wizard en la API nueva para v8
[MOD] Se reescribe la vista para hacerla mas dinamica
[MOD] Modificaciones en onchanges para filtrar mejor y proponer la
informacion necesaria. Ahora con solo rellenar dos campos o obtenemos
los datos del comprobante y se propone un comprobante para relacionar si es
que se encuentra uno solo que coincida con los datos
[FIX] Se soluciona un bug que no escribia la fecha de facturacion que
traia de la AFIP al sincronizar.

sebastiken committed:
Modificaciones datos demo para correr testing

[MOD] Se modifican y adecuan los datos de demo para que se pueda lanzar
el testing sin problemas desde linea de comandos para este modulo

sebastiken committed:
Modificacion en rutina de test

sebastiken committed:
Testing para wsfe_sinchronize_voucher

[ADD] Se agregan rutinas de testing para wizard
wsfe.sinchronize.voucher. Se utiliza mock y se arma una estructura que
permite seguir creando rutinas de testing sin necesidad de hablar
directamente con el Web Service de AFIP
[MOD] Se modifican datos de demo de WSFE para poder lanzar las rutinas
de testing correctamente

sebastiken committed:
Wizard sincronizacion masiva WSFE. Testing.

[ADD] Nuevo wizard para sincronizar todos los comprobantes de un punto
de venta y una denominacion determinadas.
[ADD] Testing para este nuevo wizard.

sebastiken committed:
Merge branch 'wsfe_tests' into sebastiken-8.0

sebastiken committed:
Mergeo a mano de wsfe.py por unas lineas mal mergeadas por git

sbidoul committed:
[ADD] addons table in README.md

sebastiken committed:
Fix Wizard sincro masiva con comprobantes iguales

[FIX] Si se daba la situacion de tener que sincronizar cuando hay dos o
mas comprobantes del mismo cliente y mismos importes, no sincronizaba
porque encontraba mas de uno y los desestimaba.

sbidoul committed:
[UPD] prefix versions with 8.0

sebastiken committed:
Traducciones actualizadas.

sebastiken committed:
Merge branch '8.0' of https://github.com/sebastiken/l10n-argentina into sebastiken-8.0

sebastiken committed:
Merge branch 'sebastiken-8.0' into 8.0

ssaid committed:
[l10n_ar_point_of_sale] More info in sale exception

fiscal_position not defined in sale order

sebastiken committed:
Merge pull request #23 from ssaid/8.0

[l10n_ar_point_of_sale] More info in sale exception

sebastiken committed:
Fix wizard sale_line_invoice

sebastiken committed:
Fix wizard sale_line_invoice II

sebastiken committed:
Utilizamos campo local de Fiscal Position

sebastiken committed:
Clase para comunicacion con WSFEX

[ADD] Clase de comunicacion con WSFEX
[ADD] Tests para WSFEX y WSFE
[MOD] Se modifica la estructura de directorios para incluir los tests

sebastiken committed:
Objeto wsfex.config

[ADD] Se agrega modelo y vistas para wsfex.config. Logica para la
creacion del wsaa.ta correspondiente
[MOD] wsfe_demo.xml y wsfe_data.xml para incluir datos para wsfex.config
[FIX] En tests por problema con el conftest.py

sebastiken committed:
Modelos para tablas de parametros WSFEX

[ADD] Se agregan modelos para los parametros que se pueden obtener desde
el WSFEX
[MOD] Se modifica la vista de wsfex.config para agregar los parametros y
el mapeo a objetos de OpenERP
[ADD] Se agregan algunos tests mas.
[FIX] Error al retornar resultado de las funciones de WSFEX

sebastiken committed:
Agregado parametro Tipo Exportacion

[ADD] Se agrega modelo y cambio de vista para parametro Tipo Exportacion
que se obtiene de la comunicacion con la AFIP

sebastiken committed:
Bug fix en funcion FEXResult_LastCMP

sebastiken committed:
Agregado de campos a factura de exportacion

[ADD] Se agregan campos para saber si es de Exportacion, Tipo de
Exportacion, Destino Pais, CUIT Pais, Permisos de embarque
[ADD] Modelo de permisos de embarque
[MOD] Se modifica la vista de factura

sebastiken committed:
Separacion por configuracion.

[MOD] Se modifica el codigo para separar mejor las funciones entre WSFE
y WSFEX. Por ejemplo, la funcion de preparacion de detalles y la de
parseo del resultado.
[ADD] Campo incoterm_id a account.invoice
[MOD] Se puede setear a mano el numero de factura y se utiliza ese.
Igualmente, se mantiene el chequeo entre el siguiente numero del sistema
y el de la AFIP si no esta en homologacion.

sebastiken committed:
Detalle de Requests para facturas exportacion.

[ADD] Campo wsfex.request.detail. Creacion y vista.

sebastiken committed:
No se permite cancelar facturas electronicas

sebastiken committed:
Comprobantes asociados en Factura Exportacion

[ADD] Logica al hacer Refund para manejar los comprobantes asociados.

sebastiken committed:
Bug fix en action_number por variable local no inicializada

sebastiken committed:
Factura Exportacion funcional, migrada a v8

[FIX] Se corrigen bugs en la validacion
[MIG] Se migran varias funciones a la api nueva

sebastiken committed:
Bug fixes en WSFE. Agregado WSFE Request en account.invoice

sebastiken committed:
Agregado ACL para WSFEX

sebastiken committed:
Merge pull request #26 from sebastiken/wsfex

Factura Exportación v8

Se migra lo hecho en la v7 sobre factura electrónica de exportación a la v8.

gabrielo77 committed:
[ADD] Add module l10n_ar_base_country_state

sebastiken committed:
Merge pull request #29 from gabrielo77/8.0-add-state-module

[ADD] Add module l10n_ar_base_country_state

OCA-git-bot committed:
[UPD] addons table in README.md

sebastiken committed:
Fix cuando se aplican retenciones negativas (NC)

gabrielo77 committed:
[FIX] Fix issue #31

gabrielo77 committed:
Merge branch '8.0' of github.com:gabrielo77/l10n-argentina into 8.0
